### DISCUSSION TITLE
RFC: pause game process when no active connections

### DISCUSSION THREAD

**Discussion by cdaringe** [mbround18/valheim-docker#878] [Ideas]
# problem

the container burns a lot of cpu idling, even if no one is playing

# discussion

1. docker image hosts a tiny tiny itty bitty little proxy that fwds packets to the game server
2. when no connections occur for X duration, the game PID is _stopped_ (not killed! stopped/paused)
3. when connection is attempted and we're in the stopped state, the PID is resumed

outcome: mo' betta use of resources!

**Comment by cdaringe** [comment]
chatgpt seemed to have a coherent and decent strategy, albeit i'm not sure how it would work with UDP vs TCP: https://chatgpt.com/share/9feef185-8469-45f3-96bf-ce76c8a6dfcc

**Comment by cdaringe** [comment]
oh man, this is actually _pretty great_. no proxy needed! just a background script using tcpdump!

```sh
check_udp_traffic() {
    # Capture packets for a short period and count them
    local packet_count
    packet_count=$(echo $(timeout 30  tcpdump -c 1 -i any udp port 2456  2>/dev/null) | grep UDP  | wc -l)
    if [ "$packet_count" -gt 0 ]; then
        return 0  # Traffic detected
    else
        return 1  # No traffic
    fi
}
```

i had no idea tcpdump was so rad. i adapted some the chatgpt stuff into this script and tested it--works great!

is some integration like this something you'd be willing to consider?

**Reply by mbround18** [reply]
Since the handler is rust id prefer this in the rust code as part of a job that checks for udp traffic and that code could be duplicated for your backup case. Would you mind submitting a pr? I'm cool if you use gpt just be sure to write tests. If not lmk and I can add this as a feature request to work on.

**Reply by mbround18** [reply]
@cdaringe ive created a custom gpt just for rust development if you are interested  https://chatgpt.com/g/g-eUaVpqgDX-rustscout its what i use to help me update files in this repo

**Reply by cdaringe** [reply]
i tried today. rust isn't my day-to-day, but i've done enough to be dangerous. really, i failed. i may give it another swing: https://github.com/mbround18/valheim-docker/pull/880
